spec/lib/govuk_logging_spec.rb

```diff
@@ -71,22 +74,66 @@ def app
+      it "logs errors thrown by the application with govuk_request_id" do
+        stub_const("GdsApi::GovukHeaders", GovukHeaders)
         GovukLogging.configure
         get "/error"
-        $stderr.rewind
-        lines = $stderr.read.split("\n")
+        fake_stdout.rewind
+        lines = fake_stdout.read.split("\n")
         expect(lines).to include(/default exception/)
         error_log_line = lines.find { |log| log.match?(/default exception/) }
         expect(error_log_line).not_to be_empty
+
         error_log_json = JSON.parse(error_log_line)
         expect(error_log_json).to match(hash_including(
-                                          "exception_class" => "StandardError",
-                                          "exception_message" => "default exception",
+                                          "govuk_request_id" => "some-value",
                                         ))
-        expect(error_log_json).to have_key("stacktrace")
-        expect(error_log_json["stacktrace"]).to be_a(Array)
+
+        error_log_json_msg = error_log_json["message"]
+        expect(error_log_json_msg).to match(hash_including(
+                                              "exception_class" => "StandardError",
+                                              "exception_message" => "default exception",
+                                            ))
+        expect(error_log_json_msg).to have_key("stacktrace")
+        expect(error_log_json_msg["stacktrace"]).to be_a(Array)
+      end
+
+      it "logs errors thrown by the application with no govuk_request_id" do
+        GovukLogging.configure
+        get "/error"
+        fake_stdout.rewind
+        lines = fake_stdout.read.split("\n")
+        expect(lines).to include(/default exception/)
+        error_log_line = lines.find { |log| log.match?(/default exception/) }
+        expect(error_log_line).not_to be_empty
+        error_log_json = JSON.parse(error_log_line)
+        error_log_json_msg = error_log_json["message"]
+        expect(error_log_json_msg).to match(hash_including(
+                                              "exception_class" => "StandardError",
+                                              "exception_message" => "default exception",
+                                            ))
+        expect(error_log_json_msg).to have_key("stacktrace")
+        expect(error_log_json_msg["stacktrace"]).to be_a(Array)
+      end
+
+      it "logs to stdout in JSON format with govuk_request_id" do
+        stub_const("GdsApi::GovukHeaders", GovukHeaders)
+        GovukLogging.configure
+        logger = Rails.logger
+        logger.info("test default log entry")
+        fake_stdout.rewind
+        log_line = fake_stdout.read
+        log_json = JSON.parse(log_line)
+
+        expect(log_json).to include("message" => "test default log entry")
+        expect(log_json).to include("govuk_request_id" => "some-value")
       end
     end
   end
 end
+
+class GovukHeaders
```

Comment:
Doing this will create a global constant of this. You should instead create an anonymous one in your test that can be used.

```ruby
let(:govuk_headers_class) do 
  Class.new do
    def self.headers
    { govuk_request_id: "some-value" }
    end
  end 
end

it "logs to stdout in JSON format with govuk_request_id" do
  stub_const("GdsApi::GovukHeaders", govuk_headers_class)
  # ...
```

Reply:
I understand you can delete this now.